tests/broker/test_matrix_broker.py

```diff
@@ -18,21 +19,205 @@
+    assert not hasattr(test_broker, "mutex_queue")
+    assert not hasattr(test_broker, "device_queue")
+    assert not hasattr(test_broker, "broadcast_queue")
+    assert not hasattr(test_broker, "replication_queue")
+
+    # call _init_queues
+    test_broker._init_queues(None)
+
+    # verify that the broker now has queues
+    assert hasattr(test_broker, "mutex_queue")
+    assert hasattr(test_broker, "device_queue")
+    assert hasattr(test_broker, "broadcast_queue")
+    assert hasattr(test_broker, "replication_queue")
+
+    # verify that the broker queues' room_id match the environment's room_id
+    assert test_broker.mutex_queue.room_id == environment_room_id
+    assert test_broker.device_queue.room_id == environment_room_id
+    assert test_broker.broadcast_queue.room_id == environment_room_id
+    assert test_broker.replication_queue.room_id == environment_room_id
+
+
+async def test_matrix_broker_init_queues_no_room_id_use_broker_id():
+    """
+    Tests that if None is given as a room ID, the broker's room id is used
+    """
+
+    # create a MatrixBroker object
+    test_broker = MatrixBroker()
+
+    # set the broker's room id to a custom id
+    broker_room_id = "test_room_id"
+    test_broker.room_id = broker_room_id
+
+    # verify that there are no existing queues in the broker
+    assert not hasattr(test_broker, "mutex_queue")
+    assert not hasattr(test_broker, "device_queue")
+    assert not hasattr(test_broker, "broadcast_queue")
+    assert not hasattr(test_broker, "replication_queue")
+
+    # call _init_queues
+    test_broker._init_queues(None)
+
+    # verify that the broker now has queues
+    assert hasattr(test_broker, "mutex_queue")
+    assert hasattr(test_broker, "device_queue")
+    assert hasattr(test_broker, "broadcast_queue")
+    assert hasattr(test_broker, "replication_queue")
+
+    # verify that the broker queues' room_id match the environment's room_id
+    assert test_broker.mutex_queue.room_id == broker_room_id
+    assert test_broker.device_queue.room_id == broker_room_id
+    assert test_broker.broadcast_queue.room_id == broker_room_id
+    assert test_broker.replication_queue.room_id == broker_room_id
+
+
+async def test_matrix_broker_with_result_backend_not_instance(test_matrix_broker):
+    """
+    Tests that an exception is raised if an object is passed to with_result_backend
+    that is not a MatrixResultBackend object
+    """
+
+    # create a MatrixBroker object from a fixture
+    test_broker = await test_matrix_broker()
+    # create a generic mock object that is not a MatrixResultBackend
+    mock_backend = MagicMock()
+
+    # call with_result_backend to raise an exception
+    with pytest.raises(
+        Exception, match="result_backend must be an instance of MatrixResultBackend"
+    ):
+        test_broker.with_result_backend(mock_backend)
+
+
+async def test_matrix_broker_with_result_backend_is_instance(test_matrix_broker):
```

Comment:
Instead of mocking with_result_backend like that, just call `test_broker.with_result_backend(mock_backend)` and assert that it returns a broker instance with the result_backend set to the passed instance.